### PR TITLE
what's-new for removal of analysis interpolation module

### PR DESCRIPTION
**Pull request by corinnebosley**

### CODE REVIEW

**Comment by corinnebosley** [comment]
@dkillick You want me to pop the word 'deprecated' in there to cover our backs?

**Comment by corinnebosley** [comment]
@dkillick I did it anyway.

**Comment by dkillick** [comment]
> You want me to pop the word 'deprecated' in there to cover our backs?

Sure, why not 😉

File: docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_remove-analysis-interpolation.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+* The deprecated module ```iris.analysis.interpolate``` has been removed, along with the following functions:
```
Comment (dkillick) [inline docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_remove-analysis-interpolation.txt]: Why not add "deprecated" before "functions" as well...

Also, `Linear1DExtrapolator` was a class and not a function, so you should probably turn the last bit of this sentence into "the following deprecated classes and functions:".

File: docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_remove-analysis-interpolation.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    * ```iris.analysis.linear```
+    * ```iris.analysis.all_points_in_range```
```
Comment (dkillick) [inline docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_remove-analysis-interpolation.txt]: This is a method in `Linear1DExtrapolator`, which technically was never deprecated (outside of being part of the deprecated class), so it should not be in this list.

File: docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_remove-analysis-interpolation.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    * ```iris.analysis.nearest_neighbour_data_value```
+    * ```iris.analysis.regrid```
+    * ```iris.analysis.extract_nearest_neighbour```
+    * ```iris.analysis.nearest_neighbour_indices```
+    * ```iris.analysis.Linear1dExtrapolator```
```
Comment (dkillick) [inline docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_remove-analysis-interpolation.txt]: You're missing `iris.analysis.interpolate.regrid_to_max_resolution` from this list.

Also, you need to add "`interpolate`" to each item in the list...

**Comment by corinnebosley** [comment]
@dkillick I added the [ci skip] which worked its lovely magic, so you will have to check the changes you requested.

**Pull request merged**